### DISCUSSION TITLE
Custom Getter/Setter for a Model's property

### DISCUSSION THREAD

**Discussion by psikoi** [prisma/prisma#11335] [Q&A]
I've tried looking around for this feature on both the documentation and this repo's issues, but I haven't seen any reference to it.

Does Prisma have a way to set custom getters and setters for a model's property?

Real life scenario:

I have a numeric column in my table that can go up to 4.6 billion, so I have to store it as BIGINT in Postgres, however I want to handle it in JS as a number, as it's well under the `Number` maximum limit. 

I've been trying to migrate from Sequelize to Prisma, and with Sequelize I can declare a property like so:

```js
  @Column({ type: DataType.BIGINT, get: getValue, set: setValue })
  value: number;
```

and in those get/set functions, I can manipulate the bigint string I get from Postgres into a js number, and vice versa.

---

Is there such a way to choose how a field gets stored, and gets retrieved, without using middlewares? (those can get unnecessarily complex for such a simple operation/conversion)

Thanks

**Comment by austincrim** [comment]
Hey @psikoi 👋 ,

Prisma does not support this pattern today. You are correct in assuming [middleware](https://www.prisma.io/docs/concepts/components/prisma-client/middleware) would be the current best way to solve it.

It would be great if you could open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) with more details about how you would like this to work, our product and engineering teams will pick it up from there.

Let me know if you have any more questions!

**Comment by psikoi** [comment]
This is what I ended up doing:

https://github.com/prisma/prisma/discussions/5522#discussioncomment-2761676

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
